### PR TITLE
feat(event-emitter): add `promise` method

### PR DESCRIPTION
**Pull request by nakasyou**
## Description

I added a `promise` method to TypedEventEmitter.
If you merged it, you can write code like that:
```ts
const [pin] = await client.promise('pincall')
```

## Checklist

- [x] Run tests (optional)
- [x] Add jsdoc (optional)
- [x] Test in your environment (optional)

If you feel good :), please do the contents of the checklist.

### CODE REVIEW

File: packages/client/libs/typed-event-emitter/index.ts [hunk 1]
```diff
@@ -41,4 +41,20 @@ export class TypedEventEmitter<
 
 		return this;
 	}
+
+	/**
+	 * This creates a promise that you can use for a single event.
+	 * @param event A event name
+	 */
+	public promise<E2 extends E, P extends unknown[] = Parameters<T[E2]>>(
```
Comment (EdamAme-x) [inline packages/client/libs/typed-event-emitter/index.ts:49]: ### My Proposals
- `await client.wait("pincall");`
- `await client.waitFor("pinncall");`

how about this?

Comment (EdamAme-x) [inline packages/client/libs/typed-event-emitter/index.ts:49]:
```suggestion
	public promise<E2 extends E, P = Parameters<T[E2]>>(
```

I think that this is more simple

Comment (nakasyou) [reply packages/client/libs/typed-event-emitter/index.ts:49]: Thanks I agree.

Comment (nakasyou) [reply packages/client/libs/typed-event-emitter/index.ts]: I wasn't satisfied for this naming, so your proposal looks great.
I think `wait` is simple, but probably it needs discussion.

**Pull request merged**